### ISSUE TITLE
Rounding issue

### ISSUE THREAD

**Issue by angelokisilmarino** [acarril/texresults#2]
Hi, thanks for your great work! I am having an issue with some coefficients that are not rounding as expected. I am trying to round to 3 decimal places, but the command is exporting some coefficients with 16 decimal places. It does round to 3 decimal places as expected, but fills in up to the sixteenth with 12 zeros and a 1 at the end.

**Comment by acarril** [comment]
Hi, sorry it took me so long to reply. Based on what you describe, it seems like bug induced by floating points. I'll take a closer look and report back.

**Comment by acarril** [comment]
Could you please provide a minimum working example so that I can reproduce the issue?

**Comment by angelokisilmarino** [comment]
Hi, thanks for answering. The following example reproduces the rounding issue I reported here:

`sysuse auto, clear`

`reg mpg trunk foreign`

`texresults using results.tex, texmacro(trunk) coef(trunk) unitzero replace`

**Comment by acarril** [comment]
Unfortunately I am not able to reproduce the issue you describe. When running your example above, I get
```latex
% results.tex
\newcommand{\trunk}{$-0.68$}
```
which is the expected behavior: the result has a zero in the unit position, and it is rounded to two decimal places, which is the default. I am using Stata/MP 17.0 for Mac (Intel 64-bit), revision 01 Jun 2022. Below is a screenshot of what I run and the output I get.

<img width="637" alt="image" src="https://user-images.githubusercontent.com/9773292/174897860-5a7d1d7c-6655-4027-8283-ffce13f45cb7.png">

**Comment by angelokisilmarino** [comment]
Indeed, it may be a version problem then, when I run on Stata 15, I get the output copied below. However, when I run on Stata 17, I get -0.68, just like you. I am sorry for that.

![image](https://user-images.githubusercontent.com/95148003/175037785-d74e00e1-101b-4f02-adae-f67c9c94f27d.png)

**Comment by acarril** [comment]
I see, no need to apologize! I presume it must be that the `round()` function works differently for different Stata versions. Unfortunately I don't have access to Stata 15 to try to debug this, so unless you're able to find a solution, it is unlikely that I'll be able to fix this issue on my end.

**Comment by gns36** [comment]
Hi! Thanks again for creating this package. I am having a similar rounding issue but slightly different circumstance. I am using matrices to create the lower and upper bounds on my coefficients and then also export those using texresults. However, one of my bounds also has the same rounding issue. I have tried exporting the contents from the matrix as a local, and also rounding the matrix itself. I also updated to stata18 but am still having this issue. Here is some code that recreates the issue: 

`//recreate issue using auto data 
sysuse auto, clear
reg mpg trunk foreign


//create matrices to find lower and upper bounds
mat var = e(V)
local v = var[1,1]
mat se = sqrt(var[1,1])
mat B = e(b)
mat b = B[1,1]
mat lb = b-num*se

//export results 
texresults using results.txt, texmacro(lb) result(lb[1,1]) unitzero replace 
cat results.txt`

The results show: \newcommand{\lb}{$-0.9500000000000001$} 
instead of \newcommand{\lb}{$-0.95$} 

Any help solving this issue would be greatly appreciated. Thank you for your time.
